cmd/embedded-cluster/reset.go

```diff
@@ -0,0 +1,299 @@
+	}
+	return nil
+}
+
+// getControlNodeObject fetches the controlNode object from the k8s api server
+func (h *hostInfo) getControlNodeObject(ctx context.Context) error {
+	err := h.Kclient.Get(ctx, client.ObjectKey{Name: h.Hostname}, &h.ControlNode)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *hostInfo) checkQuorumSafety(c *cli.Context) (bool, string, error) {
+	if c.Bool("yes-really-reset") {
+		return true, "", nil
+	}
+	out, err := exec.Command(k0s, "etcd", "member-list").Output()
+	if err != nil {
+		return false, "", fmt.Errorf("unable to fetch etcd member list, %w, %s", err, out)
+	}
+	etcd := etcdMembers{}
+	err = json.Unmarshal(out, &etcd)
+	if err != nil {
+		return false, "", fmt.Errorf("unable to unmarshal etcd member list, %w, %s", err, out)
+	}
+	if len(etcd.Members) < 3 {
+		return true, "", nil
+	}
+	if len(etcd.Members) == 3 {
+		return false, "cluster has 3 control-plane nodes, removing this node will cause etcd to lose quorum", nil
+	}
+	if len(etcd.Members)%2 != 0 {
+		return false, "cluster would have even number of control-plane nodes after resetting this node, this could cause etcd to become unstable", nil
+	}
+	return true, "", nil
+}
+
+// leaveEtcdcluster uses k0s to attempt to leave the etcd cluster
+func (h *hostInfo) leaveEtcdcluster() error {
+	out, err := exec.Command(k0s, "etcd", "leave").CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("unable to leave etcd cluster: %w, %s", err, string(out))
+	}
+	return nil
+}
+
+// stopK0s attempts to stop the k0s service
+func (h *hostInfo) stopAndResetK0s() error {
+	out, err := exec.Command(k0s, "stop").CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("could not stop k0s service: %w, %s", err, string(out))
+	}
+	out, err = exec.Command(k0s, "reset").CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("could not reset k0s: %w, %s", err, string(out))
+	}
+	fmt.Println("Node has been reset, please reboot to ensure transient configuration is also reset")
+	return nil
+}
+
+// newHostInfo returns a populated hostInfo struct
+func newHostInfo(ctx context.Context) (hostInfo, error) {
+	currentHost := hostInfo{}
+	// populate hostname
+	err := currentHost.getHostName()
+	if err != nil {
+		return currentHost, err
+	}
+	// set up kube client
+	err = currentHost.configureKubernetesClient()
+	if err != nil {
+		return currentHost, err
+	}
+	// fetch node object
+	err = currentHost.getNodeObject(ctx)
+	if err != nil {
+		return currentHost, err
+	}
+	// control plane only stff
+	if currentHost.isControlPlane() {
+		// fetch controlNode
+		err := currentHost.getControlNodeObject(ctx)
+		if err != nil {
+			return currentHost, err
+		}
+	}
+	return currentHost, nil
+}
+
+func checkErrPrompt(err error) bool {
+	if err == nil {
+		return true
+	}
+	fmt.Println("-----")
+	fmt.Println(err)
+	fmt.Println("-----")
+	fmt.Println("An error has occured while trying to reset this node.")
+	fmt.Println("Continuing may leave the cluster in an unexpected state")
```

Comment:
Let's be consistent with periods and use them at the end of sentences. There are a few other cases without them too.